perfkitbenchmarker/windows_virtual_machine.py

```diff
@@ -0,0 +1,306 @@
+    """Waits until VM is has booted."""
+    resp, _ = self.RemoteCommand('hostname', suppress_warning=True)
+    if self.bootable_time is None:
+      self.bootable_time = time.time()
+    if self.hostname is None:
+      self.hostname = resp[:-1]
+
+  def OnStartup(self):
+    stdout, _ = self.RemoteCommand('echo $env:TEMP')
+    self.temp_dir = os.path.join(stdout.strip(), 'pkb')
+    self.RemoteCommand('mkdir %s' % self.temp_dir)
+    self.DisableGuestFirewall()
+
+  def Install(self, package_name):
+    """Installs a PerfKit package on the VM."""
+    if ((self.is_static and not self.install_packages) or
+        not FLAGS.install_packages):
+      return
+    if package_name not in self._installed_packages:
+      package = windows_packages.PACKAGES[package_name]
+      package.Install(self)
+      self._installed_packages.add(package_name)
+
+  def Uninstall(self, package_name):
+    """Uninstalls a Perfkit package on the VM."""
+    package = windows_packages.PACKAGES[package_name]
+    if hasattr(package, 'Uninstall'):
+      package.Uninstall()
+
+  def PackageCleanup(self):
+    """Cleans up all installed packages.
+
+    Deletes the temp directory and uninstalls all PerfKit packages.
+    """
+    for package_name in self._installed_packages:
+      self.Uninstall(package_name)
+    self.RemoteCommand('rm -recurse -force %s' % self.temp_dir)
+    self.DisableGuestFirewall()
+
+  def _GetNumCpus(self):
+    """Returns the number of logical CPUs on the VM.
+
+    This method does not cache results (unlike "num_cpus").
+    """
+    stdout, _ = self.RemoteCommand(
+        'Get-WmiObject -class Win32_processor | '
+        'select -exp NumberOfLogicalProcessors')
+    return int(stdout)
+
+  def _GetTotalMemoryKb(self):
+    """Returns the amount of physical memory on the VM in Kilobytes.
+
+    This method does not cache results (unlike "total_memory_kb").
+    """
+    stdout, _ = self.RemoteCommand(
+        'Get-WmiObject -class Win32_PhysicalMemory | '
+        'select -exp Capacity')
+    return int(stdout) / 1024
+
+  def _TestReachable(self, ip):
+    """Returns True if the VM can reach the ip address and False otherwise."""
+    try:
+      self.RemoteCommand('ping -n 1 %s' % ip)
+    except errors.VirtualMachine.RemoteCommandError:
+      return False
+    return True
+
+  def DownloadFile(self, url, dest):
+    """Downloads the content at the url to the specified destination."""
+
+    command = 'Invoke-WebRequest {url} -OutFile {dest}'.format(
+        url=url, dest=dest)
+    self.RemoteCommand(command)
+
+  def UnzipFile(self, zip_file, dest):
+    """Unzips the file with the given path."""
+    command = ('Add-Type -A System.IO.Compression.FileSystem; '
+               '[IO.Compression.ZipFile]::ExtractToDirectory(\'{zip_file}\', '
+               '\'{dest}\')').format(zip_file=zip_file, dest=dest)
+    self.RemoteCommand(command)
+
+  def DisableGuestFirewall(self):
+    """Disables the guest firewall."""
```

Comment:
I think this will break compatibility with 2008 r2 - https://technet.microsoft.com/en-us/library/jj554896(v=wps.630).aspx says 2012 r2 and windows 8.1